src/libmongoc/src/mongoc/mongoc-crypt.c

```diff
@@ -1019,6 +1163,7 @@ _mongoc_crypt_new (const bson_t *kms_providers,
 
    if (!success) {
       _mongoc_crypt_destroy (crypt);
+      bson_destroy (&crypt->kms_providers);
```

Comment:
```suggestion
```

This is destroyed in `_mongo_crypt_destroy`. This may be the cause of double frees in ASAN tasks.